packages/typespec-go/src/emitter.ts

```diff
@@ -170,9 +171,41 @@ function truncateStack(stack: string, finalFrame: string): string {
   return stack;
 }
 
+/**
+ * Clean up existing generated Go files in the output directory.
+ * Removes any .go files that contain the Microsoft code generator comment.
+ * 
+ * exported for testing purposes only.
+ *
```

Comment:
If its exported then its public. We shouldn't make functions public just for testing.